public/app/features/alerting/unified/utils/misc.ts

```diff
@@ -31,7 +31,7 @@ export function createExploreLink(dataSourceName: string, query: string) {
   return createUrl(`/explore`, {
     left: JSON.stringify({
       datasource: dataSourceName,
-      queries: [{ refId: 'A', datasource: dataSourceName, expr: query }],
+      queries: [{ refId: 'A', expr: query }],
```

Comment:
We'd honestly prefer we keep the datasource reference here. Does the same bug happen if you pass in a datasource UID and go 

```
queries: [{ refId: 'A', datasource: {uid: uid}, expr: query }],
```

Reply:
Unfortunately that also seems to trigger the bug when the outer `datasource` is the name iso an identifier.